Update CI documentation following recent evolutions.

Zimmi48 opened the pull request:
<!-- Thank you for your contribution.
     Make sure you read the contributing guide and fill this template. -->


<!-- Keep what applies -->
**Kind:** documentation


<!-- If this is a bug fix, make sure the bug was reported beforehand. -->
This allows checking the two first items at #6807.

Further improvements possible.

Zimmi48 commented on dev/ci/README.md:
```diff
@@ -54,7 +50,7 @@ Add a new `ci-mydev.sh` script to [`dev/ci`](/dev/ci) (have a look at
 [`ci-basic-overlay.sh`](/dev/ci/ci-basic-overlay.sh); add the corresponding
 target to [`Makefile.ci`](/Makefile.ci); add new jobs to
-[`.travis.yml`](/.travis.yml) and [`.gitlab-ci.yml`](/.gitlab-ci.yml) so that
+[`.circleci/config.yml`](/.circleci/config.yml) and [`.gitlab-ci.yml`](/.gitlab-ci.yml) so that
```
Yes, but this PR actually goes hand in hand with my request here: https://github.com/coq/coq/pull/7436#issuecomment-388346027
That's why I felt it was time to document how to run GitLab CI on private branches.

ejgallego replied:
That's fine but I'd like to keep Travis a bit more. We can just wait to merge this PR tho.

Zimmi48 commented on dev/ci/README.md:
```diff
@@ -126,7 +129,8 @@ Currently, available artifacts are:
   architecture and Ocaml version used to build Coq.
 - the Coq documentation, in two different copies varying in the OCaml
-  version used to build Coq
+  version used to build Coq. When submitting a documentation PR, this
```
It was also my impression. Will fix this.

Zimmi48 commented:
Updated.

ejgallego commented on dev/ci/README.md:
```diff
@@ -36,10 +36,6 @@ On the condition that:
-  If this is not the case, consider adding a "lite" target that compiles just
-  part of it.
-
 In case you forget to comply with these last three conditions, we would reach
```
I'd suggest we keep a paragraph about build time, even if we don't put a concrete number; "reasonable time" should be mentioned, there are a few Coq developments that may take in the order of > 10 hours.

Zimmi48 replied:
Sounds good.

ejgallego commented on dev/ci/README.md:
```diff
@@ -71,24 +67,38 @@ When you submit a pull request (PR) on Coq GitHub repository, this will
 unless these tests pass.
 
-Currently, we have two CI platforms:
+We are currently running tests on many different platforms:
```
I feel that "many" is a bit overkill, why not just "We currently run tests on the following CI platforms:"

Zimmi48 replied:
OK

ejgallego commented on dev/ci/README.md:
```diff
@@ -71,24 +67,38 @@ When you submit a pull request (PR) on Coq GitHub repository, this will
+
+- Circle CI runs tests that are redundant with GitLab CI and will be removed
+  eventually.
+
```
This discussion belongs elsewhere but there is still a good use case for Circle: chains of long depending contribs. There, Circle does a better job. So given that Circle maintenance overhead is quite low [using Docker] I wouldn't mind keeping some jobs like that on Circle.

ejgallego replied:
Well I guess it depends how do you read the sentence, maybe the current wording is OK.

ejgallego commented on dev/ci/README.md:
```diff
@@ -97,36 +107,29 @@ so that it doesn't, or provide a branch fixing these developments (or at
-GitLab is set up to use the "build artifact" feature to avoid
+GitLab CI is set up to use the "build artifact" feature to avoid
 rebuilding Coq. In one job, Coq is built with `./configure -prefix
 install` and `make install` is run, then the `install` directory
```
AFAICT the prefix here is `_install_ci`.

Zimmi48 replied:
Indeed.

ejgallego commented:
Question on CODEOWNERS, shouldn't the rule be for `dev/ci/*` instead of only for the `.sh` part?

I didn't add an owner to Docker as I assumed the were covering the whole directory, but now I see this is not the case.

SkySkimmer commented:
>shared runner status

What link is that?

Gaëtan Gilbert

On 12/05/2018 14:21, Emilio Jesús Gallego Arias wrote:
> While we are at it, I think that adding a link to the pipelines page and 
> a link to the shared runner status would be good.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub 
> <https://github.com/coq/coq/pull/7482#issuecomment-388551462>, or mute 
> the thread 
> <https://github.com/notifications/unsubscribe-auth/ACWQ7G7JAk8891VSArLAmAQQtY280yVMks5txtPcgaJpZM4T7bGP>.
>

ejgallego commented:
> shared runner status
> 
> What link is that?

The one with the build graphs you posted the other day on Gitter

ejgallego commented:
@Zimmi48 let me know if you want to address the comments, and to what extent; basically I am fine merging in the state you consider appropriate once you confirm you have read the suggestions.

Zimmi48 commented:
@ejgallego I've addressed your comments so that it can be merged now (including putting back the link to `.travis.yml`). We can update this document further when it is needed. Wrt the CODEOWNER thing, I will prepare an update.

ejgallego commented:
Thanks, @SkySkimmer , could you re-post the gitlab status link here [my gitter search skills are blah]

SkySkimmer commented:
https://monitor.gitlab.net/dashboard/db/ci
